policybot/pkg/testresults/testresults.go

```diff
@@ -0,0 +1,378 @@
+		onePull.StartTime = tm
+
+	}
+	return onePull, nil
+}
+
+func (prt *PrResultTester) getInformationFromCloneFile(pref string, onePull *store.TestResult) (*store.TestResult, error) {
+	client := prt.Client
+	bucket := client.Bucket(prt.bucketName)
+	obj := bucket.Object(pref + "/clone-records.json")
+	scope.Infof("read clone")
+	rdr, err := obj.NewReader(prt.ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rdr.Close()
+	slurp, err := ioutil.ReadAll(rdr)
+	if err != nil {
+		return nil, err
+	}
+	s := string(slurp)
+	dec := json.NewDecoder(strings.NewReader(s))
+
+	_, err = dec.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	for dec.More() {
+		var record cloneRecord
+		err := dec.Decode(&record)
+		if err != nil {
+			return nil, err
+		}
+
+		refs := record.Refs
+		pulls := refs.Pulls
+		pull := pulls[0]
+		sha := pull.Sha
+		baseSha := refs.BaseSha
+
+		failed := record.Failed
+
+		onePull.Sha = sha
+		onePull.BaseSha = baseSha
+		onePull.CloneFailed = failed
+
+	}
+	return onePull, nil
+}
+
+/*
+ * GetShaAndPassStatus function return the status of test passing, clone failure, sha number, base sha for each test run under each test suite for the given pr.
+ * Client: client used to get buckets and objects from google cloud storage.
+ * TestSlice: a slice of Tests objects containing all tests and the path to folder for each test run for the test under such pr.
+ * Return a map of test suite name -- pr number -- run number -- ForEachRun objects.
+ */
+func (prt *PrResultTester) getShaAndPassStatus(testSlice []tests, orgID string, repoID string) ([]*store.TestResult, error) {
+	var allTestRuns = []*store.TestResult{}
+
+	for _, test := range testSlice {
+		testName := test.Name
+
+		prefs := test.Prs
+
+		for _, pref := range prefs {
+
+			var onePull = &store.TestResult{}
+			onePull.OrgID = orgID
+			onePull.RepoID = repoID
+			onePull.TestName = testName
+			onePull.RunPath = pref
+
+			var err error
+			onePull, err = prt.getInformationFromCloneFile(pref, onePull)
+			if err != nil {
+				return nil, err
+			}
+
+			onePull, err = prt.getInformationFromStartedFile(pref, onePull)
+			if err != nil {
+				return nil, err
+			}
+
+			onePull, err = prt.getInformationFromFinishedFile(pref, onePull)
+			if err != nil {
+				return nil, err
+			}
+
+			prefSplit := strings.Split(pref, "/")
+
+			runNo, errr := strconv.ParseInt(prefSplit[len(prefSplit)-1], 10, 64)
+			if errr != nil {
+				return nil, errr
+			}
+			onePull.RunNum = runNo
+			prNo, newError := strconv.ParseInt(prefSplit[len(prefSplit)-3], 10, 64)
+			if newError != nil {
+				return nil, newError
+			}
+			onePull.PrNum = prNo
+			allTestRuns = append(allTestRuns, onePull)
+
+		}
+	}
+	return allTestRuns, nil
+}
+
+/*
+ * Read in gcs the folder of the given pr number and write the result of each test runs into a slice of TestFlake struct.
+ */
+func (prt *PrResultTester) CheckTestResultsForPr(prNum int64, orgID string, repoID string) ([]*store.TestResult, error) {
```

Comment:
While PR Numbers are integers, is it useful to store them as such?  Do we ever need to compare PR numbers, increment, decrement, or perform other arithmetic on them?